### PR TITLE
Update form-vars-sub to use data-amp-replace and value attrs

### PR DESCRIPTION
**Pull request by mkhatib**
A [follow up to this discussion](https://github.com/ampproject/amphtml/pull/6486#discussion_r90979604).

Let me know if this looks good, will update tests.

### CODE REVIEW

File: src/service/url-replacements-impl.js [hunk 1]
```diff
@@ -645,6 +646,81 @@ export class UrlReplacements {
...
+        }));
+  }
+
+  /**
+   * Expands an input element value attribute with variable substituted.
+   * @param {!HTMLInputElement} element
+   * @return {string} Replaced string for testing
+   */
+  expandInputValueSync(element) {
+    dev().assert(element.tagName == 'INPUT' &&
```
Comment (cramforce) [inline src/service/url-replacements-impl.js]: Refactor to remove code duplication with Async function.

Comment (mkhatib) [inline src/service/url-replacements-impl.js]: Done

**Comment by mkhatib** [comment]
PTAL 👀

File: src/service/url-replacements-impl.js [hunk 2]
```diff
@@ -645,6 +646,85 @@ export class UrlReplacements {
...
+   */
+  getWhitelistForElement_(element, opt_supportedReplacement) {
+    const whitelist = element.getAttribute('data-amp-replace');
+    if (!whitelist) {
+      return;
+    }
+    const requestedReplacements = {};
+    whitelist.trim().split(/\s+/).forEach(replacement => {
+      if (!opt_supportedReplacement ||
+          (opt_supportedReplacement &&
```
Comment (jridgewell) [inline src/service/url-replacements-impl.js]: This check is useless?

Comment (mkhatib) [inline src/service/url-replacements-impl.js]: That's true. this can be dropped. 👍

**Pull request merged**